homeassistant/components/modbus/modbus.py

```diff
@@ -166,8 +169,27 @@ def _log_error(self, exception_error: ModbusException, error_state=True):
             _LOGGER.error(log_text)
             self._in_error = error_state
 
+    def control_scan_delay(self, name, raw_scan_interval):
```

Comment:
I suggest we move the logic to a custom voluptuous validator instead. We have access to all the scan intervals and timeout settings in the config. We can modify the set config with the validator and/or fail the validation. See eg:

https://github.com/home-assistant/core/blob/ec08256ff0b3dd62861aa6468619e9ce4ae4b57a/homeassistant/components/mysensors/__init__.py#L61-L70

Reply:
Even better idea, I will do that.

Reply:
My preference is to live with the workarund and I remove the old style configuration for 2021.7 (4 month after first warning)